components/board/genericlinux/board.go

```diff
@@ -241,13 +240,11 @@ func (b *sysfsBoard) reconfigureInterrupts(newConf *Config) error {
 			if err := oldInterrupt.Close(); err != nil {
 				return err // This should never happen, but the linter worries anyway.
 			}
-			if pinInt, err := strconv.Atoi(oldInterrupt.config.Pin); err == nil {
-				if newGpioConfig, ok := b.gpioMappings[pinInt]; ok {
-					// See gpio.go for createGpioPin.
-					b.gpios[oldInterrupt.config.Pin] = b.createGpioPin(newGpioConfig)
-				}
+			if newGpioConfig, ok := b.gpioMappings[oldInterrupt.config.Pin]; ok {
+				// See gpio.go for createGpioPin.
+				b.gpios[oldInterrupt.config.Pin] = b.createGpioPin(newGpioConfig)
 			} else {
-				b.logger.Warnf("Unable to reinterpret old interrupt pin '%s' as GPIO, ignoring.",
+				b.logger.Warnf("Unable to create old interrupt pin '%s' as GPIO, ignoring.",
```

Comment:
What does it mean to create an old interrupt pin as a GPIO? I think the word change is wrong, or perhaps even that this entire `else` clause should be removed. To get here, we need an digital interrupt which has disappeared during reconfiguration, whose GPIO pin has a name we no longer recognize. When could that happen? 

I suspect this should never happen, meaning that if it does happen we should probably log at the error level. and the thing to log sounds more like `"old interrupt pin was on nonexistent GPIO pin '%s'"` or something like that.

(I might have made a mistake: if this could happen sometimes, I wonder if not logging anything here is the right choice. Note that in the previous version, this logging was for a failed call to `strconv.Atoi`, and there was no logging at all if `b.gpioMappings` didn't contain the old interrupt pin.)